File: Backend/Travle.Model/Responses/RecommendationResponse.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// The current user's personalized recommendations (04 §3). <see cref="IsColdStart"/> is true when the
    /// user has too little signal for content-based scoring and the list is a pure-popularity fallback,
    /// labeled as such for the UI. Each <see cref="RecommendationItem"/> carries a light destination card
    /// (thumbnail only, rule 12), its blended score, and a human-readable reason.
    /// </summary>
    public class RecommendationResponse
    {
        public List<RecommendationItem> Items { get; set; } = new();

        /// <summary>True when the list is the cold-start popularity fallback rather than content-based.</summary>
        public bool IsColdStart { get; set; }
    }

    /// <summary>One recommended destination: the light card DTO plus its score and explanation (04 §3).</summary>
    public class RecommendationItem
    {
        public DestinationResponse Destination { get; set; } = null!;

        /// <summary>The final blended score behind the ranking (content + popularity), surfaced for transparency.</summary>
        public double Score { get; set; }

        /// <summary>Human-readable "why" — e.g. "Because you're interested in Nature".</summary>
        public string Reason { get; set; } = string.Empty;
    }
}

File: UI/travle_core/lib/src/models/recommendation_response.dart
import 'package:json_annotation/json_annotation.dart';

import 'recommendation_item.dart';

part 'recommendation_response.g.dart';

/// The current user's personalized recommendations (mirrors the backend
/// `RecommendationResponse`). [isColdStart] is true when the user has too little
/// signal for content-based scoring and the list is a pure-popularity fallback,
/// which the UI labels honestly.
@JsonSerializable()
class RecommendationResponse {
  RecommendationResponse({
    required this.items,
    required this.isColdStart,
  });

  final List<RecommendationItem> items;
  final bool isColdStart;

  factory RecommendationResponse.fromJson(Map<String, dynamic> json) =>
      _$RecommendationResponseFromJson(json);

  Map<String, dynamic> toJson() => _$RecommendationResponseToJson(this);
}
